packages/opencensus-exporter-stackdriver/src/stackdriver-monitoring.ts

```diff
@@ -0,0 +1,234 @@
+            const scopes = ['https://www.googleapis.com/auth/cloud-platform'];
+            authClient = authClient.createScoped(scopes);
+          }
+
+          return authClient;
+        })
+        .catch((err) => {
+          err.message = `authorize error: ${err.message}`;
+          throw (err);
+        });
+  }
+
+  /**
+   * Creates a Stackdriver TimeSeries from a given view and metric value.
+   * @param view The view to get TimeSeries information from
+   * @param measurement The measurement to get TimeSeries information from
+   */
+  private createTimeSeriesData(view: View, measurement: Measurement):
+      TimeSeries {
+    const aggregationData = view.getSnapshot(measurement.tags);
+
+    const resourceLabels:
+        {[key: string]: string} = {project_id: this.projectId};
+
+    // For non Sum Aggregations, the end time should be the same as the start
+    // time.
+    const endTime = (new Date(aggregationData.timestamp)).toISOString();
+    const startTime = view.aggregation === AggregationType.SUM ?
+        (new Date(view.startTime)).toISOString() :
+        endTime;
+
+    let value;
+    if (view.measure.type === MeasureType.INT64) {
+      value = {int64Value: measurement.value.toString()};
+    } else if (aggregationData.type === AggregationType.DISTRIBUTION) {
+      value = {distributionValue: this.createDistribution(aggregationData)};
+    } else {
+      value = {doubleValue: measurement.value};
+    }
+
+    return {
+      metric: {
+        type: `custom.googleapis.com/${view.name}`,
+        labels: measurement.tags
+      },
+      resource: {type: 'global', labels: resourceLabels},
+      metricKind: this.createMetricKind(view.aggregation),
+      valueType: this.createValueType(view),
+      points: [{interval: {startTime, endTime}, value}]
+    };
+  }
+
+  /**
+   * Formats an OpenCensus' Distribution to Stackdriver's format.
+   * @param distribution The OpenCensus Distribution Data
+   */
+  private createDistribution(distribution: DistributionData): Distribution {
+    return {
+      count: distribution.count.toString(),
+      mean: distribution.mean,
+      sumOfSquaredDeviation: distribution.sumSquaredDeviations,
+      range: {min: distribution.min, max: distribution.max},
+      bucketOptions: {
+        explicitBuckets:
+            {bounds: this.getBucketBoundaries(distribution.buckets)}
+      },
+      bucketCounts: this.getBucketCounts(distribution.buckets)
+    };
+  }
+
+  /**
+   * Gets the bucket boundaries in an monotonicaly increasing order.
+   * @param buckets The bucket list to get the boundaries from
+   */
+  private getBucketBoundaries(buckets: Bucket[]): number[] {
+    return [...buckets.map(bucket => bucket.lowBoundary), Infinity].sort(
+        (boundary1, boundary2) => {
+          return boundary1 < boundary2 ? -1 : 1;
+        });
+  }
+
+  private getBucketCounts(buckets: Bucket[]): number[] {
+    return buckets
+        .sort((bucket1, bucket2) => {
```

Comment:
I'm not sure that making buckets sorted as precondition is a great idea. Since `buckets` is public, anyone (just like I was doing) can mutate it, making it harder to guarantee it will always be sorted. We should trust the user won't change the order.

Thanks for remembering about the cloning!